File: server/Microservices/BookingService/BookingService.Domain/Models/SessionSeatsModel.cs
namespace BookingService.Domain.Models;

public class SessionSeatsModel
{
	public Guid Id { get; private set; }
	public Guid SessionId { get; private set; }
	public IList<SeatModel> AvailableSeats { get; private set; } = [];
	public IList<SeatModel> ReservedSeats { get; private set; } = [];
	public DateTime UpdatedAt { get; private set; }

	public SessionSeatsModel(Guid id, Guid sessionId, IList<SeatModel> availableSeats, IList<SeatModel> reservedSeats, DateTime updatedAt)
	{
		Id = id;
		SessionId = sessionId;
		AvailableSeats = availableSeats;
		ReservedSeats = reservedSeats;
		UpdatedAt = updatedAt;
	}
}

File: src/mobile-flutter/lib/data/models/session/session_seats_model.dart
// lib/data/models/session/session_seats_model.dart
import '../../../domain/entities/session/session_seats.dart';
import 'seat_model.dart';

class SessionSeatsModel extends SessionSeats {
  const SessionSeatsModel({
    required String sessionId,
    required List<SeatModel> reservedSeats,
  }) : super(sessionId: sessionId, reservedSeats: reservedSeats);

  factory SessionSeatsModel.fromJson(Map<String, dynamic> json) {
    final List<dynamic> seatsJson = json['reservedSeats'] ?? [];
    final List<SeatModel> seats =
        seatsJson.map((seat) => SeatModel.fromJson(seat)).toList();

    return SessionSeatsModel(
      sessionId: json['sessionId'],
      reservedSeats: seats,
    );
  }
}
